html/embed/index.js

```diff
@@ -139,4 +159,76 @@ function handleTsParam(ts) {
   }
 
   return ts;
-}
\ No newline at end of file
+}
+
+/**
+ * Waits for a given element to be in the DOM and returns it.
+ * Wait is based on `requestAnimationFrame`: timeout is approximately 60 seconds (60 x 60 frames per seconds).
+ *
+ * Takes a function querying the DOM for a single element as an argument
+ */
+async function waitForElement(selectorFunction) {
+  const maxPauseSeconds = 60
+  let tries = maxPauseSeconds * 60;  // we expect a repaint rate of ~60 times a second, per https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame
+  let elem = null;
+
+  while (!elem && tries > 0) {
+    // Sleep efficiently until the next repaint
+    const pause = await new Promise(resolve => requestAnimationFrame(resolve));
+    cancelAnimationFrame(pause);
+
+    // Look for the target element
+    try {
+      elem = selectorFunction();
+    }
+    catch (err) {
+      if (!err.message.includes('null')) {
+        throw err;
+      }
+      tries -= 1;
+    }
+  }
+
+  if (elem) {
+    return elem;
+  }
+
+  throw new Error("Timed out");
+}
+
+/**
+ * Async helper function for handling `overrideElementAttribute` messages.
+ * Posts `overrideElementAttribute` back to the parent frame on failure.
+ */
+async function overrideElementAttribute(origin, player, parent, selector, attributeName, attributeContents){
```

Comment:
Something you could do here is use [@typedef](https://jsdoc.app/tags-typedef.html) to create a loose type definition that may simplify your code a bit.

Here's an example:
```javascript
/**
 * @typedef {Object} HTMLAttributeOverride
 * @property {string} selector - Foo bar baz
 * @property {string} attributeName - Foo bar baz
 * @property {strimg} attributeContents - Foo bar baz
 */
```

This allows you to then simplify `overrideElementAttribute()` quite a bit, and avoid having to repeat yourself.

```javascript
/**
 * Async helper function for handling `overrideElementAttribute` messages.
 * Posts `overrideElementAttribute` back to the parent frame on failure.
 * 
 * @param {string} origin - Foo bar baz
 * @param {HTMLElement} player - Foo bar baz
 * @param {HTMLElement} parent - Foo bar baz
 * @param {HTMLAttributeOverride} overrideElementAttribute
 * @returns {Promise<void>}
 */
async function overrideElementAttribute(origin, player, parent, overrideElementAttribute){
  const { selector, attributeName, attributeContents } = overrideElementAttribute;
}
```